uw-frame-components/portal/widgets/services.js

```diff
@@ -13,35 +13,49 @@ define(['angular'], function(angular) {
      */
     var getSingleWidgetData = function getSingleWidgetData(fname) {
       return $http.get(SERVICE_LOC.widgetApi.entry + fname + '.json')
-        .then(function(result) {
+        .then(function (result) {
           if (result.data.entry.layoutObject != undefined) {
             return result.data.entry.layoutObject;
           }
         })
-        .catch(function(error) {
+        .catch(function (error) {
           $log.warn('Error getting marketplace entry for ' + fname);
           $log.error(error);
           return getErrorPage(fname);
-      });
+        });
     };
 
-    var getErrorPage = function(fname) {
-      var layoutObject={title:fname, 
-        mdIcon:"md-help", widgetType: "generic",
-      widgetConfig:{additionalText: "Please contact your help desk if you feel you should be able to access this content"},
-        widgetTemplate:"<i class='fa fa-exclamation-triangle fa-3x pull-left' style='color: #b70101;'></i><span style='color: #898989;'>You do not have permission to access this content. If you feel this is an error, please contact your help desk.</span>"}
-      return layoutObject;
-    }
+    /**
+     * Display error message if a user doesn't have permission to see/use the requested widget
+     * @param fname
+     * @returns {Object} Special error-case widget configuration
+     */
+    var getErrorPage = function (fname) {
+      return {
+        title: fname,
+        mdIcon: "help",
+        widgetType: "generic",
+        widgetConfig: {
+          additionalText: "Please contact your help desk if you feel you should be able to access this content"
+        },
+        widgetTemplate: "<div class='overlay__maintenance-mode'>"
+          + "<div class='maintenance-content'>"
+          + "<p><md-icon class='md-warn'>warning</md-icon></p>"
+          + "<p>You do not have permission to access this content. If you feel this is an error, please contact your help desk.</p>"
```

Comment:
Is lack of permission the only way to catch an error at line 21? What happens if the `fname` doesn't map to anything? The underlying portlet-definition is expired? MyUW's web services are unavailable because the `/portal` context collapsed under its own bloated weight?

Opportunity to differentiate amongst failure modalities, or in the meantime to use more generic error messaging for this default "something bad happened but dunno what" case?